From: Haggai Abramonvsky <hagaya@mellanox.com>
Date: Thu, 4 Jun 2015 19:30:39 +0300
Subject: [PATCH 072/117] net/mlx5_core: Check the return bitmask when querying
 ISSI
Patch-mainline: v4.2-rc1
Git-commit: e74a1db03326ecdc7c3f54a581b197a26ce5eebe
References: bsc#923036 FATE#318772

The determination of the supported ISSI versions should be conditioned
on the returned mask, and not only on the return status of the query
ISSI command, fix that.

Signed-off-by: Haggai Abramovsky <hagaya@mellanox.com>
Signed-off-by: Majd Dibbiny <majd@mellanox.com>
Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Hadar Hen Zion <hadarh@mellanox.com>
Acked-by: Cho, Yu-Chen <acho@suse.com>
---
 drivers/net/ethernet/mellanox/mlx5/core/main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/mellanox/mlx5/core/main.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/main.c
@@ -654,7 +654,7 @@ static int mlx5_core_set_issi(struct mlx
 		dev->issi = 1;
 
 		return 0;
-	} else if (sup_issi & (1 << 0)) {
+	} else if (sup_issi & (1 << 0) || !sup_issi) {
 		return 0;
 	}
 
